kura/org.eclipse.kura.protocol.can.test/src/main/java/org/eclipse/kura/protocol/can/test/CanSocketTest.java

```diff
@@ -117,6 +101,37 @@ public void updated(Map<String, Object> properties) {
         }
     }
 
+    @Before
+    public void setup() {
+        Assume.assumeFalse(IS_MAC);
+    }
+
+    @Test
+    public void startCanTestThread() {
+        if (this.m_pollThread != null) {
+            this.m_pollThread.interrupt();
+            try {
+                this.m_pollThread.join(100);
+            } catch (InterruptedException e) {
+                // Ignore
+            }
+            this.m_pollThread = null;
+        }
+
+        this.m_pollThread = new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                if (CanSocketTest.this.m_canConnection != null && !IS_MAC) {
```

Comment:
`CanSocketTest.this.thread_done` does not need to be a class variable. I've changed the code accordingly.